### PR TITLE
Ups json api labels

### PR DESCRIPTION
**Pull request by davecandlescience**

### CODE REVIEW

**Review by mamhoff — APPROVED** [review]
Looks great!

File: lib/friendly_shipping/services/ups_json/generate_labels_payload.rb [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+              payload[:ShipmentRequest][:Shipment][:InvoiceLineTotal] = {
+                CurrencyCode: 'USD',
+                MonetaryValue: total_value.to_f.to_s
+              }
+            end
+
+            contents_description = shipment.packages.flat_map do |package|
+              package.items.map(&:description)
+            end.compact.uniq.join(', ').slice(0, 50)
+            payload[:ShipmentRequest][:Shipment][:Description] = contents_description
```
Comment (pelargir) [inline lib/friendly_shipping/services/ups_json/generate_labels_payload.rb:108]: Didn't you already handle this above on lines 11-13 and 23?

Comment (davecandlescience) [reply lib/friendly_shipping/services/ups_json/generate_labels_payload.rb:108]: Yes, nice catch.

**Review by pelargir — COMMENTED** [review]
I have some suggestions. Let me know if you'd like to discuss.

File: lib/friendly_shipping/services/ups_json/generate_labels_payload.rb [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+          payload[:ShipmentRequest][:Shipment][:ShipFrom]
+
+          if options.customer_context.present?
+            payload[:ShipmentRequest][:Request][:TransactionReference] = { CustomerContext: options.customer_context }
+          end
+
+          if options.shipper || options.return_service_code
+            origin = options.return_service_code ? shipment.destination : shipment.origin
+            payload[:ShipmentRequest][:Shipment][:ShipFrom] = GenerateAddressHash.call(location: origin, international: international?(shipment))
+            payload[:ShipmentRequest][:Shipment][:ShipTo] = GenerateAddressHash.call(location: shipment.origin, international: international?(shipment))
```
[the remark this answers is not in the document]
Comment (davecandlescience) [reply lib/friendly_shipping/services/ups_json/generate_labels_payload.rb]: I'm not following it either! This was copied from the xml api and seems wrong to me too.

**Review by pelargir — APPROVED** [review]

**Pull request merged**